drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,398 @@
+		}
+	}
+
+	return rc;
+}
+
+static int auxdisplay_serlcd_init(const struct device *dev)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+
+	if (!device_is_ready(config->bus.bus)) {
+		return -ENODEV;
+	}
+
+	auxdisplay_serlcd_clear(dev);
+
+	return 0;
+}
+
+static const struct auxdisplay_driver_api auxdisplay_serlcd_auxdisplay_api = {
+	.display_on = auxdisplay_serlcd_display_on,
+	.display_off = auxdisplay_serlcd_display_off,
+	.cursor_set_enabled = auxdisplay_serlcd_cursor_set_enabled,
+	.position_blinking_set_enabled = auxdisplay_serlcd_position_blinking_set_enabled,
+	.cursor_position_set = auxdisplay_serlcd_cursor_position_set,
+	.cursor_position_get = auxdisplay_serlcd_cursor_position_get,
+	.capabilities_get = auxdisplay_serlcd_capabilities_get,
+	.clear = auxdisplay_serlcd_clear,
+	.custom_character_set = auxdisplay_serlcd_custom_character_set,
+	.write = auxdisplay_serlcd_write,
+};
+
+#define AUXDISPLAY_SERLCD_INST(inst)                                                               \
+	static const struct auxdisplay_serlcd_config auxdisplay_serlcd_config_##inst = {           \
+		.capabilities =                                                                    \
+			{                                                                          \
+				.columns = DT_INST_PROP(inst, columns),                            \
+				.rows = DT_INST_PROP(inst, rows),                                  \
+				.mode = 0,                                                         \
+				.brightness.minimum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,              \
+				.brightness.maximum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,              \
+				.backlight.minimum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,               \
+				.backlight.maximum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,               \
+				.custom_characters = 8,                                            \
+				.custom_character_width = 5,                                       \
+				.custom_character_height = 8,                                      \
+			},                                                                         \
+		.bus = I2C_DT_SPEC_INST_GET(inst)};                                                \
+                                                                                                   \
+	static struct auxdisplay_serlcd_data auxdisplay_serlcd_data_##inst = {                     \
+		.display_state = {.padding = 0,                                                    \
+				  .power = true,                                                   \
+				  .cursor = false,                                                 \
+				  .blinking = false},                                              \
+		.cursor_x = 0,                                                                     \
+		.cursor_y = 0,                                                                     \
+	};                                                                                         \
```

Comment:
initializing this at build time causes the data structure to get a copy in flash rather than be zero-initialized automatically, it's not a big deal in this case as that's just 3 bytes but I think it's good practice to avoid that since you only need to initialize one bit from here and the rest is zero anyway

I suggest dropping the static initialization and just set the `power` bit at init time

Reply:
Done